File: matlab/solutions/ml_3_classification/exo7.m
W = zeros(p,k);
Elist = [];
tau = .01;
niter = 500;
for i=1:niter
    W = W - tau * nablaE(W);
    Elist(i) = E(W);
end
clf;
plot(1:niter, Elist, 'LineWidth', 2);
axis tight;
SetAR(1/2); 

File: julia/NtSolutions/ml_3_classification/exo7.jl
W = zeros(p,k)
Elist = []
tau = .01
niter = 500
for i=1:niter
    W = W - tau * nablaE(W)
    append!(Elist, E(W))
end
plot(1:niter, Elist, lw=2)
axis("tight")
